pkg/routing/inmemory/inmemory.go

```diff
@@ -56,7 +57,9 @@ func (r *NodeInfoStore) Add(ctx context.Context, nodeInfo model.NodeInfo) error
 	} else {
 		var engines []model.Engine
 		if nodeInfo.ComputeNodeInfo != nil {
-			engines = append(engines, nodeInfo.ComputeNodeInfo.ExecutionEngines...)
+			for _, e := range nodeInfo.ComputeNodeInfo.ExecutionEngines {
+				engines = append(engines, e)
+			}
```

Comment:
The previous version was one recommended by the linter. Is there a reason it isn't suitable?